Exercise #2 - JS Basics. Part 2.

Humarik opened the pull request:
This pull request implements tasks 1-6. 

# Практическая работа 2 - Основы JavaScript. Часть 2.

## Задание 1

Написать функцию, которая принимает либо число, либо строку и в зависимости от типа переданного аргумента возвращает `'number'` или `'string'`. Если аргумент не число/строка функция возвращает `undefined`.

## Задание 2

Написать функцию, которая будет пробегать по массиву и выводить в консоль каждый элемент массива и общее число элементов.

## Задание 3

Написать функцию, которая посчитает и выведет в консоль количество четных и нечетных элементов в массиве. Если в массиве есть нулевой элемент, то он учитывается и выводится отдельно. Возвращаемое значение -- массив вида `[<кол-во четных>, <кол-во нечетных>, <кол-во нулей>]`.
При выполнении задания необходимо учесть, что массив может содержать не только числа, но и, например, знаки, null и тд.

```js
f([1, 2, 3, 4]);
// четных: 2; нечетных: 2
// [2, 2, 0]

f([1, 2, 3, 0]);
// четных: 1; нечетных: 2; нуль: 1
// [1, 2, 1]
```

## Задание 4

Написать функцию, которая проверяет одинаковые ли элементы в массиве и возвращает `true` или `false`.
Речь идет не о двух рядом стоящих одинаковых элементах, а обо всех.

## Задание 5

Написать функцию, которая возвращает максимальное значение в массиве.

Humarik committed:
fix unit 2 task 1

Humarik committed:
fix unit 2 task 1

Humarik committed:
added js-basics-part2 task-02

*added function parseArray
*that accepts array and display each item and array length

Humarik committed:
added js-basics-part2 task-02

*added function parseArray
*that accepts array and display each item and array length

Humarik committed:
added js-basics-part-2 task-03

*added function that iterates each element and sorts them in new array

Humarik committed:
added js-basics-part-2 task-03

*added function that iterates each element and sorts them in new array

Humarik committed:
added ex2_js-basics-part2 task-04

*added function that accepts array and checks each elements
*if function found alike elements, this one changes flag on true

Humarik committed:
added ex2_js-basics-part2 task-04

*added function that accepts array and checks each elements
*if function found alike elements, this one changes flag on true

Humarik committed:
added js-basics-part-2 task-05

*added function that finds max number in array

Humarik committed:
added js-basics-part-2 task-05

*added function that finds max number in array

Humarik committed:
added js-basics-part2 task-06

*Added function that checks number and shows prime number or not

Humarik committed:
added js-basics-part2 task-06

*Added function that checks number and shows prime number or not

Humarik committed:
modified README.md

*added word 'выполнено' for git home work

Humarik committed:
modified README.md

*added word 'выполнено' for git home work

Humarik committed:
fixed mistake task-06

*have been deleted empty space in function

Humarik committed:
fixed mistake task-06

*have been deleted empty space in function

Humarik committed:
fixed mistake task-03

* have been deleted extra 'let' for variables odd, zero

Humarik committed:
fixed mistake task-03

* have been deleted extra 'let' for variables odd, zero

Humarik committed:
fixed mistake task-04

*have been deleted excess console.log

Humarik committed:
fixed mistake task-04

*have been deleted excess console.log

aneelia commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,18 @@
+function counter(array) {
+    let resultArray = [];
```
resultArray вообще не нужен, можно без него обойтись

aneelia commented on src/ex2_js-basics-part2/task-03.js:
```diff
@@ -0,0 +1,18 @@
+function counter(array) {
```
функция не отвечает на вопрос, что она делает

aneelia commented on src/ex2_js-basics-part2/task-04.js:
```diff
@@ -0,0 +1,16 @@
+function checking(array) {
```
checking что

aneelia commented on src/ex2_js-basics-part2/task-04.js:
```diff
@@ -0,0 +1,16 @@
+function checking(array) {
+    let x = 0;
```
можно объединить два объявления переменных, используя 1 Let

Humarik committed:
Fix tasks 3,4,6

These tasks have been fixed for better reading.
Repeating declaration 'let' has been deleted.
Function names have been changed.
Cycle has been changed and deleted variable 'result' in task-03.

Humarik committed:
Fix tasks 3,4,6

These tasks have been fixed for better reading.
Repeating declaration 'let' has been deleted.
Function names have been changed.
Cycle has been changed and deleted variable 'result' in task-03.

Humarik committed:
Fix task-02

'var' has been chenged on 'let' and param of function on 'array'

Humarik committed:
merge

Humarik reviewed (commented):
_It seems, I needed to update my current branch because, maybe, I did something wrong and when I wanted to make push tasks, I got mistake from git that I have to use git pull. After this I read some articles and did something. xD It seems, it was 'upstream' my branch. After this I was surprised, when git pull get working and I got merge conflict. I was sooo scary. But, I just took my last changed commit that was in local host and make new commit 'merge' and my last commit 'Fix task-02' has been pushed too_ 
**I apologize for this, I'm just newbie :)**

aneelia commented:
Dont worry
